### PR TITLE
Make wowza configuration tenant-specific

### PR DESCRIPTION
**Pull request by KatrinIhler**
This PR moves a couple streaming configuration keys around to make it tenant-specific, allowing it to configure streaming for multi-tenant installations.
It also removes the "adaptive" from the class names, modules and so on to make naming more consistent.
Additionally, as RTMP is rarely used for on-demand streaming nowadays, the support was removed since this makes the code much easier to understand.

If you've got a Wowza installation, it would be great if you could test this!

### CODE REVIEW

File: etc/custom.properties
Comment (lkiesow) [inline etc/custom.properties]: If this should be the new default value, let's make it an actual default:

```suggestion
# Default: *
#org.opencastproject.publish.streaming.formats=*
```

File: etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg
Comment (lkiesow) [inline etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg]: What's the default? Is the default still empty and the service deactivated if this is empty/blank?

File: etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg
Comment (lkiesow) [inline etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg]:
```suggestion
# Tenant specific port for Wowza, usually 1935
```

File: etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg
Comment (lkiesow) [inline etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg]: Is there a default?

File: modules/distribution-service-api/src/main/java/org/opencastproject/distribution/api/StreamingDistributionService.java
Comment (lkiesow) [inline modules/distribution-service-api/src/main/java/org/opencastproject/distribution/api/StreamingDistributionService.java:42]: What is this method for? Short JavaDoc? ;-p

File: modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java
Comment (lkiesow) [inline modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java]: 1. this should be a ConcurrentHashMap
2. You shouldn't set it first and then fill it, but fill it  and only at the end update the whole configuration at once

File: modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java
Comment (lkiesow) [inline modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java]: Wouldn't that be a valid configuration?
This is basically warning people that they aren't using streaming which doesn't make sense, does it?
I suggest dropping this to debug level.

**Review by lkiesow — CHANGES_REQUESTED** [review]

**Comment by mliradelc** [comment]
Great work with this PR!! I'm starting to testing it right now, I need only to finish the configuration from the wowza site to put it to work.

I have a question, since opencast can sign the wowza url, in theory should work the `org.opencastproject.security.urlsigning.provider.impl.WowzaUrlSigningProvider.cfg` without issues right? because you can set a key per url.

**Review by mliradelc — CHANGES_REQUESTED** [review]
This part in the documentation:

```markdown
4. Enter a name for the new application. You **must** use the same application name you have configured
in `$KARAF/etc/custom.properties` (for instance: *opencast-engage*) in `$KARAF/etc/custom.properties` (for instance: *opencast-engage*)
```

This is from the previous version and doesn't explain very well (Says name, but not what specific variable to change)

~~With also this in mind, Does each Wowza server needs to use the same application name?~~ 
Ex:
wowza.server.com/opencast-engage
wowza.server2.com/opencast-engage
wowza.server3.com/opencast-engage
... 

~~Or it can be different application names also??~~


Nevermind I figure out myself, the names correspond to the names of the applications you put in the URL.

File: modules/distribution-service-api/src/main/java/org/opencastproject/distribution/api/StreamingDistributionService.java
Comment (KatrinIhler) [reply modules/distribution-service-api/src/main/java/org/opencastproject/distribution/api/StreamingDistributionService.java:42]: Whoops, forgot.

File: modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java
Comment (KatrinIhler) [reply modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java]: Will do.

File: modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java
Comment (KatrinIhler) [reply modules/distribution-service-streaming-wowza/src/main/java/org/opencastproject/distribution/streaming/wowza/WowzaStreamingDistributionService.java]: Done.

File: etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg
Comment (KatrinIhler) [reply etc/org.opencastproject.distribution.streaming.wowza.WowzaStreamingDistributionService.cfg]: Yep, I'll add that to the comment.

**Comment by KatrinIhler** [comment]
> Great work with this PR!! I'm starting to testing it right now, I need only to finish the configuration from the wowza site to put it to work.
> 
> I have a question, since opencast can sign the wowza url, in theory should work the `org.opencastproject.security.urlsigning.provider.impl.WowzaUrlSigningProvider.cfg` without issues right? because you can set a key per url.

Yeah, I think that still works, you'll need to set a key per application.

**Comment by KatrinIhler** [comment]
@lkiesow I think I addressed everything. Thanks for taking a look at it! :)

**Comment by mliradelc** [comment]
Tested. Worked without any issues. Wowza player is able to play from different streams.

I didn't test the streaming security

**Review by mliradelc — APPROVED** [review]

**Comment by KatrinIhler** [comment]
@mliradelc Since you're a committer, you can merge this if you're fine with it.

**Pull request merged**
